lib/stitches/circular_fill.py

```diff
@@ -0,0 +1,88 @@
+from shapely import geometry as shgeo
+
+from ..stitch_plan import Stitch
+from ..utils.geometry import reverse_line_string
+from .auto_fill import (build_fill_stitch_graph, build_travel_graph,
+                        collapse_sequential_outline_edges, fallback,
+                        find_stitch_path, graph_is_valid, travel)
+from .contour_fill import _make_fermat_spiral
+
+
+def circular_fill(shape,
+                  angle,
+                  row_spacing,
+                  num_staggers,
+                  max_stitch_length,
+                  running_stitch_length,
+                  running_stitch_tolerance,
+                  skip_last,
+                  starting_point,
+                  ending_point,
+                  underpath,
+                  target
+                  ):
+
+    # get furthest distance of the target point to a shape border
+    # so we know how many circles we will need
+    distance = shape.hausdorff_distance(target) + 1
+    radius = row_spacing
+    center = shgeo.Point(target)
+
+    circles = []
+    # add a small inner circle to make sure that the spiral ends close to the center
+    circles.append(shgeo.LineString(center.buffer(0.1).exterior.coords))
+    while distance > radius:
+        circles.append(shgeo.LineString(center.buffer(radius).exterior.coords))
+        radius += row_spacing
+    circles.reverse()
+
+    # Use double spiral from contour fill (we don't want to get stuck in the middle of the spiral)
+    double_spiral = _make_fermat_spiral(circles, max_stitch_length, circles[0].coords[0])
```

Comment:
Why not underpath instead? It seems simpler.

Reply:
I'm a fan of the Fermat spiral choice, personally.  I don't think people will always be using circular fill as a 100% coverage fill stitch.  If it's not full coverage, then underpath won't be hidden.

That said, we should actually try sewing this out with a row spacing of 0.25mm.  It may be that the inward spiral of the first half of the fermat spiral causes unacceptable puckering.  In that case we would need to sew from the center out, and it might be worthwhile to give the user the option of having stitching actually start in the center (with the expectation that they would trim first).